### ISSUE TITLE
InvalidCast false positive: Htmlable in a Blade raw-echo ({!! !!})

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#1196]
## Summary

Psalm reports `InvalidCast` when a Blade `{!! !!}` raw-echo receives an expression typed as `Illuminate\Contracts\Support\Htmlable`, because the compiled `{!! !!}` echo requires a string-castable value and `Htmlable` only declares `toHtml(): string`, not `__toString()`.

## Reproduction

```php
/** @var \Illuminate\Contracts\Pagination\LengthAwarePaginator $paginator */
$links = $paginator->links(); // returns Illuminate\Contracts\Support\Htmlable
```

```blade
{!! $links !!}
```

Compiles to:

```php
<?php echo $links; ?>
```

## Error

```
ERROR: InvalidCast
Illuminate\Contracts\Support\Htmlable cannot be cast to string (see https://psalm.dev/103)
    <?php echo $links; ?>
```

## Why this is a false positive

`Illuminate\View\Compilers\BladeCompiler` compiles `{!! $expr !!}` to a plain `echo $expr;`. At runtime, Blade/Laravel's view rendering pipeline (and PHP's own `echo`) does **not** require `__toString()` here in the way Psalm assumes for this specific class: `Htmlable`-typed values returned from framework APIs like `Paginator::links()` are, in practice, always concrete classes (e.g. `Illuminate\Support\HtmlString`) that also implement `Stringable`/`__toString()`. The interface `Htmlable` itself doesn't declare `__toString()`, but nothing in the Laravel core ever produces an `Htmlable` instance without one, so echoing it always works at runtime; Psalm's static check is more conservative than the guaranteed runtime contract.

## Suggested fix

One of:
1. Treat a value typed as `Illuminate\Contracts\Support\Htmlable` (or any type also implementing `Illuminate\Contracts\Support\Renderable`/similar Laravel view-rendering interfaces) as string-echoable inside a Blade raw-echo (`{!! !!}`) context, since Blade's compiled output is a straight `echo`.
2. Alternatively, refine the stub for `Illuminate\Pagination\AbstractPaginator::links()` (and similar `Htmlable`-returning APIs) to return a concrete `Illuminate\Support\HtmlString` (which does implement `__toString()`) instead of the bare `Htmlable` interface, since that's what's actually returned at runtime.

## Environment

- `psalm/plugin-laravel`: dev-master
- Found while running Psalm taint/type analysis over compiled `.cache/views/*.php` Blade output in a Laravel 12 project.

**Comment by alies-dev** [comment]
Converted to the repository internal ideas list (`.alias/issues.md`) so GitHub issues stay focused on active work. Reopen or recreate this when it is prioritized.
